api/src/main/java/org/projectnessie/tools/catalog/migration/api/CatalogMigrator.java

```diff
@@ -0,0 +1,233 @@
+
+  /** Enable the stacktrace in logs in case of failures. */
+  @Value.Default
+  public boolean enableStacktrace() {
+    return false;
+  }
+
+  @Value.Check
+  void check() {
+    if (!(targetCatalog() instanceof SupportsNamespaces)) {
+      throw new UnsupportedOperationException(
+          String.format(
+              "target catalog %s doesn't implement SupportsNamespaces to create missing namespaces.",
+              targetCatalog().name()));
+    }
+
+    if (deleteEntriesFromSourceCatalog() && sourceCatalog() instanceof HadoopCatalog) {
+      throw new UnsupportedOperationException(
+          "Source catalog is a Hadoop catalog and it doesn't support deleting the table entries just from the catalog. Please configure `deleteEntriesFromSourceCatalog` as `false`");
+    }
+  }
+
+  private static final Logger LOG = LoggerFactory.getLogger(CatalogMigrator.class);
+  private final ImmutableCatalogMigrationResult.Builder resultBuilder =
+      ImmutableCatalogMigrationResult.builder();
+  private final Set<Namespace> processedNamespaces = new HashSet<>();
+
+  /**
+   * Get the table identifiers which matches the regular expression pattern input from all the
+   * namespaces.
+   *
+   * @param identifierRegex regular expression pattern. If null, fetches all the table identifiers
+   *     from all the namespaces.
+   * @return Set of table identifiers.
+   */
+  public Set<TableIdentifier> getMatchingTableIdentifiers(String identifierRegex) {
+    Catalog sourceCatalog = sourceCatalog();
+    if (!(sourceCatalog instanceof SupportsNamespaces)) {
+      throw new UnsupportedOperationException(
+          String.format(
+              "source catalog %s doesn't implement SupportsNamespaces to list all namespaces.",
+              sourceCatalog.name()));
+    }
+    LOG.info("Collecting all the namespaces from source catalog...");
+    Set<Namespace> namespaces = new LinkedHashSet<>();
+    getAllNamespacesFromSourceCatalog(Namespace.empty(), namespaces);
+
+    Predicate<TableIdentifier> matchedIdentifiersPredicate;
+    if (identifierRegex == null) {
+      LOG.info("Collecting all the tables from all the namespaces of source catalog...");
+      matchedIdentifiersPredicate = tableIdentifier -> true;
+    } else {
+      LOG.info(
+          "Collecting all the tables from all the namespaces of source catalog"
+              + " which matches the regex pattern:{}",
+          identifierRegex);
+      Pattern pattern = Pattern.compile(identifierRegex);
+      matchedIdentifiersPredicate =
+          tableIdentifier -> pattern.matcher(tableIdentifier.toString()).matches();
+    }
+    return namespaces.stream()
+        .flatMap(
+            namespace -> {
+              try {
+                return sourceCatalog.listTables(namespace).stream()
+                    .filter(matchedIdentifiersPredicate);
+              } catch (Exception exception) {
+                if (namespace.isEmpty()) {
```

Comment:
It can be a legit error that needs to be re-thrown, so this check is not enough.

Reply:
Instead of catching all the exception. Now just catching `NoSuchNamespaceException` and `IllegalArgumentException`  and ignoring them only is a default namespace.

Reply:
Why `IllegalArgumentException`?

Reply:
Because some catalogs like `HiveCatalog` returns `IllegalArgumentException` instead of `NoSuchNamespaceException` while listing tables from an empty namespace. 

https://github.com/apache/iceberg/blob/b756a5da3559be9deee9cdc674c4bf36ac291231/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java#L115